### PR TITLE
#2611 auto implementing change requests edit project form warn about required links

### LINKED ISSUE #2611: [Auto-Implementing Change Requests] - Edit Project Form Warn about Required Links
### Description

When you edit a project, there are required links you must include. The form will not submit without them, but it will not inform you why its not submitting

### Acceptance Criteria

There should be an error message clearly identifying the issue

### Proposed Solution

Do not allow the create change request button to be pressed, along with an error message

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Zwendle**
## Changes

Implemented yup schema so that its errors for required links pop up below the URL boxes in the project edit page

## Notes

- any http link will cause the error to disappear even if it doesn't match the correct box (ie: putting https://youtube.com for the Confluence URL will cause the "Invalid URL" error to disappear)
- the URL boxes are shifted up due to the error text so they no longer align with the link type boxes (looks a little funky)

## Screenshots

Screenshot 2024-07-02 at 13 38 55
Screenshot 2024-07-02 at 13 39 47
Screenshot 2024-07-02 at 13 40 13

## To Do

- [ ] check URL types to make sure they match the link type in the left boxes (if needed)
- [ ] adjust the placement of the link type boxes when errors show up (if needed)

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #2611

### CODE REVIEW

File: src/frontend/src/pages/ProjectDetailPage/ProjectForm/ProjectForm.tsx [hunk 1]
```diff
@@ -137,7 +137,7 @@ const ProjectFormContainer: React.FC<ProjectFormContainerProps> = ({
               <Box display="inline-flex" alignItems="center">
                 <Tooltip
                   title={
-                    <Typography fontSize={'16px'}>
+                    <Typography fontSize={'15px'}>
```
Comment (vsp05) [inline src/frontend/src/pages/ProjectDetailPage/ProjectForm/ProjectForm.tsx]: Just out of curiosity, why did you change the font size here?

**Review by vsp05 — CHANGES_REQUESTED** [review]
looks good, except for the part where the boxes shift up when the errormessage is displayed ... I'm not really sure how to fix that, but I guess you should play around with the sx in the linkseditview.tsx and maybe that'll do it?

File: src/frontend/src/pages/ProjectDetailPage/ProjectForm/ProjectForm.tsx (hunk 1 above)
Comment (Zwendle) [reply src/frontend/src/pages/ProjectDetailPage/ProjectForm/ProjectForm.tsx]: O that was just to make the second commit idk my vscode's kinda weird

Comment (Zwendle) [reply src/frontend/src/pages/ProjectDetailPage/ProjectForm/ProjectForm.tsx]: Changed it back to 16 in the most recent commit tho

**Review by vsp05 — APPROVED** [review]
LGTM!

**Review by walker-sean — APPROVED** [review]
LGTM

**Pull request merged**